source/develop/release-management/features/network/editable-network-names.html.md

```diff
@@ -0,0 +1,94 @@
+wiki_last_updated: 2017-10-23
+feature_name: Editable Network Names
+feature_modules: engine,api
+feature_status: Proposed
+---
+
+# Editable Network Names
+
+### Summary
+
+Allow users to edit network names even while attached to hosts.
+
+## Owner
+
+*   Name: Leon Goldberg (lgoldber)
+
+<!-- -->
+
+*   Email: <lgoldber@redhat.com>
+
+## Detailed Description
+
+Currently, network names are only editable while not being attached to any host. This is primarily due to the mapping
+between the network names to the VDSM (interface) names being immutable; once a network is attached to a host, a linux
+bridge valid VDSM name is generated by engine, and a mapping of the network name to VDSM name is stored in the database.
+
+This RFE proposes 2 changes to this functionality that will allow network names to be modifiable even while
+attached to hosts:
+
+1) Upon network creation, let users decide whether they'd like the network's name to be editable*:
+    - If editable, the network's VDSM name is set to be dervied from the network's GUID (onXXX...) regardless
+      of whether the network's name could be used as VDSM name. This allows the name to be edited without any
+      changes to VDSM name on the host side.
+    - If not editable, current behavior remains. VDSM name may or may not equal the network name, and further changes
+      while attached to hosts are prohibited as they may require changes to VDSM name.
+
+* How problematic would it be to only include the editable option with GUID based vdsm name as the only option? 
+From a legacy consideration point of view there will be some disparity; users will encounter
+both GUID and network name VDSM names, however this shouldn't affect functionality and will simplify both
+implementation and user experience. As VDSM name reporting/exposure on engine side is planned in this RFE as well,
+debugging will be less of an issue as well.
+
+
+2) Modification of editable state of network names of existing networks*:
+    - Editable to non-editable is prohibited
+    - Non-editable to editable is allowed
+
+* There are no technical considerations behind these decisions and are supposedly based on making debugging/user
+experience easier. Personally, I don't see it. In both scenarios both GUID and network name based VDSM name are possible:
+
+Editable (GUID VDSM name) to non-editable (GUID/network VDSM name) is prohibited
+Non-editable (GUID/network VDSM name) to editable (GUID vdsm name) is allowed
+
+In other words, the latter (non-editable to editable) isn't any more consistent than the former but is allowed. In both scenarios
+VDSM name may eventually be either the GUID or the network name variation. In a scenario where a non-editable network
+has a network name based VDSM name ("net-1") is changed to be non-editable, "net-1" will remain as VDSM name,
+while newly created networks that are editable will have a GUID based VDSM name.
+
+Either disallow state modifications or allow both, IMO.
+
```

Comment:
As most users will not use this feature, it means support much easier when VDSM name is equal to the network name.
For the minority of cases where the network need to be editable, we will need the support person or engineer to make the correlations.